### ISSUE TITLE
Explore AppSync opportunities for GraphQL use

### ISSUE THREAD

**Issue by antoniolofiego** [100DaysOfCloud/100DaysOfCloudApp#31]
Instead of building a crazy amount of API endpoints we could leverage GraphQL and appropriate resolvers to handle all the data fetching and manipulation from/to the frontend.

**Comment by antoniolofiego** [comment]
https://github.com/cdk-patterns/serverless/tree/master/the-simple-graphql-service
This is an idea of how this could be implemented.

**Comment by what-name** [comment]
Does this need to be focused on now? A REST API will do the job just as well and is much much simpler to develop for us now. This should be in "future plans". GraphQL is great, but it's overkill for this project, especially at this stage.
There's not really a clear upside of using GraphQL for our project, but multiple downsides - for example added complexity.

**Comment by antoniolofiego** [comment]
> Does this need to be focused on now? A REST API will do the job just as well and is much much simpler to develop for us now. This should be in "future plans". GraphQL is great, but it's overkill for this project, especially at this stage.
> There's not really a clear upside of using GraphQL for our project, but multiple downsides - for example added complexity.

I agree with this. As we mentioned before, this is an issue for future features and, while not currently being developed, is just here as a future flag.
